### PR TITLE
RavenDB-21751: Another inconsistency in the capacity

### PR DESCRIPTION
**Pull request by redknightlois**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-21751

### Additional description
Capacity is wrong after a .MoveTo() operation. 

### Type of change
- Bug fix

### How risky is the change?
- Low 

### Backward compatibility
- Non breaking change

### CODE REVIEW

**Review by maciejaszyk — APPROVED** [review]

**Comment by arekpalinski** [comment]
1. Please check the compilation issue.
2. Can we have a test for that?

File: src/Voron/Data/PostingLists/NativeIntegersList.cs [hunk 1]
```diff
@@ -145,13 +145,23 @@ public void SortAndRemoveDuplicatesAndRemovals()
...
+        if (Count == count)
+        {
+            Count = 0;
+            return count;
+        }
+
+        var theRestToCopy = new Span<long>(RawItems + count, Count - count);
+        theRestToCopy.CopyTo(new Span<long>(RawItems, Capacity));
+        Count = theRestToCopy.Length;
+
```
Comment (ayende) [inline src/Voron/Data/PostingLists/NativeIntegersList.cs:163]: I don't really like this - the issue is that we are now copying twice, and this is something that is used in high perf location.
Maybe better to have a *window* into this that would not impact the actual content?

Comment (redknightlois) [reply src/Voron/Data/PostingLists/NativeIntegersList.cs:163]: Could be an option. I am rewriting all this anyways with the cleanup of lists. but this is the correct behavior that has to be implemented at the level of the list itself. Those changes have to be done at the level of the caller. I already did that for FastPFor, but @maciejaszyk was investigating something that may be related to this issues, so I prepared for him a safe version.

File: src/Voron/Util/PFor/FastPForDecoder.cs [hunk 2]
```diff
@@ -104,6 +105,7 @@ public int Read(long* output, int outputCount)
         var prefixAmount = _prefixShiftAmount;
         var sharedPrefixMask = Vector256.Create<long>(_sharedPrefix);
 
+        var bigDeltaStart = 0;
         var bigDeltaOffsets = new NativeIntegersList(_allocator, -1);
```
Comment (redknightlois) [inline src/Voron/Util/PFor/FastPForDecoder.cs:109]: @ayende version without he `.First()` and `.Pop()`

**Review by ayende — APPROVED** [review]

**Comment by redknightlois** [comment]
Superseded by https://github.com/ravendb/ravendb/pull/17801

**Pull request closed without merging**